compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -378,6 +378,18 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
             for e <- betaReduce(expr) yield tpd.cpy.Block(tree)(Nil, e)
           case tpd.Inlined(_, Nil, expr) =>
             betaReduce(expr)
+          case tpd.Apply(tpd.Select(expr: Apply, nme), args) =>
+            betaReduce(expr).map { expr1 =>
+              dotc.transform.BetaReduce(
+                tpd.Apply(tpd.Select(expr1, nme), args)
+              ).withSpan(tree.span)
+            }
+          case tpd.Apply(tpd.TypeApply(tpd.Select(expr: Apply, nme), tpts), args) =>
```

Comment:
This logic is becoming a bit ad-hoc. I feel that my original design was not the best. 

I wonder if we could simplify this using a TreeMap. I have something like this in mind. Though I have not tested it.
```scala
val tree1 = new TreeMap {
  def transform(tree: Tree)(using Context): Tree = tree match {
    case tpd.Block(Nil, _) | tpd.Inlined(_, Nil, _) => super.transform(tree)
    case _: tpd.Apply | _: tpd.TypeApply  =>
      val tree1 = super.transform(tree)
      if tree.tpe.isInstanceOf[MethodicType] then tree1
      else dotc.transform.BetaReduce(tree1).withSpan(tree.span)
    case _ => tree
  }
}.transform(tree)
if tree1 eq tree then None else Some(tree1)
```

Reply:
I replaced this with a TreeMap now but I was unable to simplify the overall logic there by much (but at the very least the `Block` and `Inlined` cases are simpler)